Application/Frontend/components/FriendsColumn/FriendsTab.tsx

```diff
@@ -111,48 +110,53 @@ export function FriendsTab(props: TextInputProps) {
         };
         fetchData(); 
       }, []);
+
+          // Filter friendList based on search query
+    const filteredFriendList = friendList.filter((friend) =>
+      friend.toLowerCase().includes(searchQuery.toLowerCase())
+    );
     
-    return (
-      <Stack>
-          {/** Search Box*/}
-          <TextInput
-              style={{ width: '100%' }}  // Ensure the TextInput takes up only available space
-              radius="xl"
-              size="md"
-              placeholder="Search..."
-              leftSection={<IconSearch color="orange" stroke={1.5} />}
-              rightSection={
-                  <ActionIcon radius="xl" variant="filled">
-                      <IconArrowRight stroke={1.5} />
-                  </ActionIcon>
-              }
-          />
-      
-          {/** "All friends" text*/}
-          <Text
-              fw={500}
-              className="text-xl"
-              component="span"
-              size="xl" 
-          >
-              All friends
-          </Text>
-          {/** Users that are in the friendlist*/}
-          {
-              friendList
-                  .map((friend, index) => (
+      return (
+        <Stack>
+            {/** Search Box*/}
+            <TextInput
+                style={{ width: '100%' }}
+                radius="xl"
+                size="md"
+                placeholder="Search..."
+                leftSection={<IconSearch color="orange" stroke={1.5} />}
+                rightSection={
+                    <ActionIcon radius="xl" variant="filled">
+                        <IconArrowRight stroke={1.5} />
+                    </ActionIcon>
+                }
+                value={searchQuery}
+                onChange={(event) => setSearchQuery(event.currentTarget.value)}
```

Comment:
Excellent use of the `onChange` parameter to dynamically update the search results in real time.